core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -571,6 +323,457 @@ Expression translateCast(
     return scaleIntervalToNumber(sourceType, targetType, convert);
   }
 
+  private Expression converterToDate(RelDataType sourceType, Expression operand) {
```

Comment:
I would prefer to split this refactoring with adding the new types into two PRs, as it will make clear what is added by this PR and what is newly added code. By doing so we could better review the new code.

Let's see what are other reviewers' opinions.

Reply:
The aim of the refactoring is make sure the length of a method is smaller than 370, and  avoid complaining by checkstyle. I will split the refactoring part to a separate commit, that would clarify the PR.